scripts/changelog-check

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env ruby
+require 'open3'
+require 'optparse'
+
+CHANGELOG_REGEX = %r{^\+- (?<category>[\w ]{2,}): (?<change>[\w ]{2,})}
+PR_REGEX = %r{ \([#\d+, ]*\d+\)$}
```

Comment:
Couple thoughts:

- `[\w ]` in the change description might be a bit restrictive, excluding punctuation and such
- Do the two patterns need to be separate?
- Is the `+` in the PR character set doing anything?
- As-is, the PR regex only guarantees a number within parantheses, not necessarily a valid hash-prefixed PR

Maybe something like... ?

```suggestion
CHANGELOG_REGEX = %r{^\+- (?<category>[\w ]{2,}): (?<change>.+) \([#\d, ]+\d+\)$}
```

Reply:
Actually, the false negatives were due to some trailing spaces in the test cases I had, so I've updated the regex to:
`CHANGELOG_REGEX = %r{^\- (?<category>[\w -]{2,}): (?<change>.+) \((?<pr>(#\d+, ?)*(#\d+))\)$}`